File: proj/code-repair/mycode/clean_dataset/patch-single-line/quixbugs-java/codex-sh/kheapsort.java
--- 
+++ 
@@ -5,7 +5,8 @@
     }
 
     ArrayList<Integer> output = new ArrayList<Integer>();
-    for (Integer x : arr) {
+
+    for (Integer x : arr.subList(k, arr.size())) {
         heap.add(x);
         Integer popped = heap.poll();
         output.add(popped);

File: proj/code-repair/mycode/clean_dataset/patch-single-line/quixbugs-python/ct5/kheapsort.py
--- 
+++ 
@@ -3,7 +3,7 @@
     heap = arr[:k]
     heapq.heapify(heap)
 
-    for x in arr:
+    for x in arr[k:] :
         yield heapq.heappushpop(heap, x)
 
     while heap:
